source/common/event/timer_impl.h

```diff
@@ -16,7 +16,33 @@ namespace Event {
  */
 class TimerUtils {
 public:
-  static void millisecondsToTimeval(const std::chrono::milliseconds& d, timeval& tv);
+  /**
+   * Intended for consumption by enable(HR)Timer, this method is templated method to avoid implicit
+   * duration conversions for its input arguments. This lets us have an opportunity to check bounds
+   * before doing any conversions. When the passed in duration exceeds INT32_MAX max seconds, the
```

Comment:
Thanks. Looks good. My main feedback below.

Can you add some explanation of why we need to clip here in the first place? I.e. how do the limitations around `timeval` come into play? I think that when we clip it would be ideal to clip to the max amount allowed by `timeval`.

Reply:
Shall I add a comment that links to https://github.com/envoyproxy/envoy/pull/9229#issuecomment-562498924 for context?

Reply:
I think a summary probably that captures the key points (that comment is long ;) )

Reply:
Done in 911bac6